src/main/java/org/veupathdb/service/eda/ds/plugin/correlationassaymetadata/CorrelationAssayMetadataBipartitenetworkPlugin.java

```diff
@@ -45,19 +50,46 @@ protected boolean computeGeneratesVars() {
   }
 
   @Override
-  protected void validateVisualizationSpec(EmptyDataPluginSpec pluginSpec) throws ValidationException {
+  protected void validateVisualizationSpec(CorrelationAssayMetadataBipartitenetworkSpec pluginSpec) throws ValidationException {
     // nothing to do here
   }
 
   @Override
-  protected List<StreamSpec> getRequestedStreams(EmptyDataPluginSpec pluginSpec) {
+  protected List<StreamSpec> getRequestedStreams(CorrelationAssayMetadataBipartitenetworkSpec pluginSpec) {
     // this plugin only uses the stats result of the compute; no tabular data streams needed
     return Collections.emptyList();
   }
 
   @Override
   protected void writeResults(OutputStream out, Map<String, InputStream> dataStreams) throws IOException {
-    System.out.println("I'm writing results!");
-    // writeComputeStatsResponseToOutput(out);
+
+    CorrelationAssayMetadataStatsResponse stats = getComputeResultStats(CorrelationAssayMetadataStatsResponse.class);
+
+
+    
+    useRConnectionWithRemoteFiles(Resources.RSERVE_URL, dataStreams, connection -> {
+      
+      // Take the stats response object and write it into R
+      // as a data.frame. There's probably a much more elegant solution, please advise!
+      connection.voidEval("statsDf <- data.frame(correlationCoef = character(), data1 = character(), data2 = character(), stringsAsFactors=FALSE)");
+      for (int i = 0; i < stats.getStatistics().toArray().length; i++) {
+        CorrelationPoint point = stats.getStatistics().get(i);
+        String newRow = "list(correlationCoef='" + point.getCorrelationCoef() + "', data1='" + point.getData1() + "', data2='" + point.getData2() + "')";
+        connection.voidEval("newRow <- " + newRow);
+        connection.voidEval("statsDf <- rbind(statsDf, newRow)");
```

Comment:
im a little worried about perf here and wonder if we can find a better way to get the data in R. ill try to think about it some, or we could see if @ryanrdoherty or @dmgaldi have an idea.

Reply:
it's definitely not fast, but not as bad as i originally feared. Regardless, i think it could use a better approach so yes if any of you have any ideas im all ears!

Reply:
so have you seen this? http://rforge.net/org/doc/  there are a bunch of REXP* classes there, under org.rosuda.REngine that ive known for a while we should probably be looking at using. i wonder if it could help us here and now is a good time to learn more about it?